src/Alert/Alert.js

```diff
@@ -1,50 +1,72 @@
 import ClassNames from 'classnames'
-import React from 'react'
+import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 import { ALERT_TYPES } from '../common/constants'
 
 /**
  * Alert Component for Patternfly React
  */
-const Alert = ({ children, onDismiss, type }) => {
-  const alertClass = ClassNames({
-    alert: true,
-    'alert-danger': type === 'danger' || type === 'error',
-    'alert-warning': type === 'warning',
-    'alert-success': type === 'success',
-    'alert-info': type === 'info',
-    'alert-dismissable': onDismiss
-  })
-  const iconClass = ClassNames({
-    pficon: true,
-    'pficon-error-circle-o': type === 'danger' || type === 'error',
-    'pficon-warning-triangle-o': type === 'warning',
-    'pficon-ok': type === 'success',
-    'pficon-info': type === 'info'
-  })
+class Alert extends Component {
+  onMouseEnter = () => {
```

Comment:
Also adding some private class fields here. This may be useful to replace the `helpers.bindMethods` in other components.

Reply:
+1 fine with this if you are

Reply:
I like the using arrow functions in class properties to handle this. On the other hand, all this could be achieved by simply passing remaining props to wrapping div:

```
const Alert = ({ children, onDismiss, type, ...props }) => {
```
and in return:
```
<div
  className={alertClass}
  {...props}
 >
```
This approach would generally allow passing any props which are valid for <div> and not just limit it to onMouseEnter and onMouseLeave. What do you think?

Reply:
ugh... missed this. I guess it can remain a stateless functional component if you just pass props like Jiri mentions above. I tend to agree w/ that now.

I am fine w/ used the spread syntax or just calling out those props (and still using spread for others) like we did here:
https://github.com/patternfly/patternfly-react/blob/master/src/ToastNotification/ToastNotification.js#L42

I can go back and update ToastNotification after to be consistent...

@danseethaler thoughts?

Reply:
@priley86 love it! Check out the git diff 😍

Reply:
@priley86 I was initially looking at the `ToastNotificationList` which has an extra step of `setState` for mouse events. In that case the prop needs to be pulled out and extended to include the set state. But like you said the `ToastNotification` could be simplified. I'm happy to create a PR for that if you'd like.

Reply:
@danseethaler sure... feel free to include that in the PR if you like, or we can do a subsequent one. Whichever is easiest for you! As long as it is consistent, I am happy though ;)

Kind of O.C.D. here - so apologies in advance!